### PR TITLE
Remove unused weight property from objects

### LINKED ISSUE #9306: Remove “weight” property of objects
## Problem

Objects have a property called “weight” but from looking around the source code it seems it does not have any effect. So what's the point?

## Solutions

Remove it.

### PR DESCRIPTION
**Pull request by Wuzzy2**
Fixes #9306.

### CODE REVIEW

File: src/object_properties.cpp [hunk 1]
```diff
@@ -77,7 +76,6 @@ void ObjectProperties::serialize(std::ostream &os) const
 	writeU8(os, 4); // PROTOCOL_VERSION >= 37
 	writeU16(os, hp_max);
 	writeU8(os, physical);
-	writeF32(os, weight);
```
Comment (rubenwardy) [inline src/object_properties.cpp]: This breaks network compatibility. Do this instead:

```cpp
writeF32(os, 0.f); // Removed property (weight)
```

File: src/object_properties.cpp [hunk 2]
```diff
@@ -128,7 +126,6 @@ void ObjectProperties::deSerialize(std::istream &is)
 
 	hp_max = readU16(is);
 	physical = readU8(is);
-	weight = readF32(is);
```
Comment (rubenwardy) [inline src/object_properties.cpp]: Same here, you'll need to skip 32 bits. Not sure if there's a function for that

Comment (SmallJoker) [reply src/object_properties.cpp]: `readF32(is);`

**Comment by Wuzzy2** [comment]
Haha, I already suspected something like this. ok...

**Comment by nerzhul** [comment]
Very strange we kept it for ages :)
Okay for me

**Pull request merged**